# Páginas para programar python

1. www.mybinder.org
2. www.kaggle.com
3. www.jupyter.org/try
4. `github jobish` https://mybinder.org/v2/gh/jobish03/jobish03-Visualizacion_de_datos_para_la_toma_de_decisiones/main?filepath=welcome_binder.ipynb

Los paquetes hay que incluir en el archivo `requirements.txt`

# Andrea Garcia Carrera A00827518


# us_covid19_daily

In [3]:
import pandas as pd
import numpy as np
import matplotlib as mpl
import matplotlib.pyplot as plt

In [4]:
pip install plotly

     |████████████████████████████████| 25.3 MB 4.7 MB/s            
Note: you may need to restart the kernel to use updated packages.


In [5]:
import plotly.express as px

In [6]:
%ls

 00_00_binder.ipynb                     A5.ipynb
 00_00_primer_pasos.ipynb              'Actividad 9 (1).ipynb'
 01_01_data_types.ipynb                'Clase 1 (1).ipynb'
 01_02_read_iloc_unique_groupby.ipynb  'Clase 1.ipynb'
 02_00_data_wrangling.ipynb             datasets/
 02_01_null_display.ipynb               README.md
 02_02_basic_math.ipynb                 requirements.txt
 02_04_Linear_regression.pptx           us_covid19_daily.csv
 7_multivariantes-1.pptx                us_states_covid19_daily.csv


In [8]:
df = pd.read_csv('us_covid19_daily.csv')
df.head(10)

date  states  positive   negative  pending  hospitalizedCurrently  \
0  20201206      56  14534035  161986294  13592.0               101487.0   
1  20201205      56  14357264  160813704  13433.0               101190.0   
2  20201204      56  14146191  159286709  12714.0               101276.0   
3  20201203      56  13921360  158026052  15106.0               100755.0   
4  20201202      56  13711156  156787587  14368.0               100322.0   
5  20201201      56  13515360  155805555   8764.0                98777.0   
6  20201130      56  13338607  153863841  14883.0                96149.0   
7  20201129      56  13191020  152644033  14860.0                93357.0   
8  20201128      56  13055778  151760885  14780.0                91762.0   
9  20201127      56  12901256  150483950  14854.0                89950.0   

   hospitalizedCumulative  inIcuCurrently  inIcuCumulative  \
0                585676.0         20145.0          31946.0   
1                583420.0         19950.0          31831.0   
2                580104.0         19858.0          31608.0   
3                575452.0         19723.0          31276.0   
4                570121.0         19680.0          31038.0   
5                565093.0         19295.0          30749.0   
6                559871.0         18801.0          30469.0   
7                556477.0         18437.0          30274.0   
8                554048.0         18249.0          30109.0   
9                550644.0         18061.0          29858.0   

   onVentilatorCurrently  ...  totalTestResults          lastModified total  \
0                 7094.0  ...         204063869  2020-12-06T24:00:00Z     0   
1                 7005.0  ...         202429337  2020-12-05T24:00:00Z     0   
2                 6999.0  ...         200259581  2020-12-04T24:00:00Z     0   
3                 6867.0  ...         198404712  2020-12-03T24:00:00Z     0   
4                 6855.0  ...         196576482  2020-12-02T24:00:00Z     0   
5                 6649.0  ...         195117280  2020-12-01T24:00:00Z     0   
6                 6520.0  ...         192776284  2020-11-30T24:00:00Z     0   
7                 6245.0  ...         191173031  2020-11-29T24:00:00Z     0   
8                 6148.0  ...         189883061  2020-11-28T24:00:00Z     0   
9                 6030.0  ...         188173495  2020-11-27T24:00:00Z     0   

   posNeg  deathIncrease  hospitalizedIncrease negativeIncrease  \
0       0           1138                  2256          1172590   
1       0           2445                  3316          1526995   
2       0           2563                  4652          1260657   
3       0           2706                  5331          1238465   
4       0           2733                  5028           982032   
5       0           2473                  5222          1941714   
6       0           1136                  3394          1219808   
7       0            803                  2429           883148   
8       0           1245                  3404          1276935   
9       0           1372                  3499          1504431   

   positiveIncrease  totalTestResultsIncrease  \
0            176771                   1634532   
1            211073                   2169756   
2            224831                   1854869   
3            210204                   1828230   
4            195796                   1459202   
5            176753                   2340996   
6            147587                   1603253   
7            135242                   1289970   
8            154522                   1709566   
9            193725                   2126288   

                                       hash  
0  9cf16504f91958e803a2197daf8c2528a4eddc18  
1  6249216c5f097c94ce33a811dab011a483a42404  
2  ae30ea088584335ba4d57ee927f8dbda6add74db  
3  0f253d185ecb336cdd18a4c61996eda1b7eef13b  
4  477c17b6302d0485195e77ecf8270a974f7a3c82  
5  f15ac17558a7a0296208371cfddab34c032e5bc8  
6  6ab94db6e497c8

## 1. Cual es la variable critica?

In [9]:
df.loc[:,['positive']]

positive
0    14534035
1    14357264
2    14146191
3    13921360
4    13711156
..        ...
315         0
316         0
317         0
318         0
319         0

[320 rows x 1 columns]

### La variable critica que voy a utilizar para este proyecto va a ser la de positive porque es la que marca los casos positivos de covid en estados unidos y es la que me va  ayudar a encontrar los resultados que buesco analizar

## 2. Cuantas variables hay y que tipos son?

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 320 entries, 0 to 319
Data columns (total 25 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   date                      320 non-null    int64  
 1   states                    320 non-null    int64  
 2   positive                  320 non-null    int64  
 3   negative                  320 non-null    int64  
 4   pending                   278 non-null    float64
 5   hospitalizedCurrently     265 non-null    float64
 6   hospitalizedCumulative    278 non-null    float64
 7   inIcuCurrently            256 non-null    float64
 8   inIcuCumulative           257 non-null    float64
 9   onVentilatorCurrently     257 non-null    float64
 10  onVentilatorCumulative    250 non-null    float64
 11  recovered                 257 non-null    float64
 12  dateChecked               320 non-null    object 
 13  death                     301 non-null    float64
 14  hospitaliz

### En esta base de datos hay un total de 25 variablesy son 320 observaciones. En esta base solo hay tres tipos de datos que son float64, int64 y object. La mayoria de las variables son tipo float.

In [11]:
pip install seaborn 

     |████████████████████████████████| 292 kB 4.9 MB/s            
Note: you may need to restart the kernel to use updated packages.


In [12]:
import seaborn as sns

## 3. Histograma de variable critica


In [14]:
sns.dispolt(df.positive)
plt.title('Casos positivos en Estados Unidos')

AttributeError: module 'seaborn' has no attribute 'dispolt'